### PR TITLE
add backend logic to disable/enable course subscriptions

### PR DESCRIPTION
**Pull request by shannon-heh**
- store `disabled_courses` as set of course ids in admin table in DB
- set up `disable_course`, `enable_course` endpoints
- create `add_disabled_course`, `remove_disabled_course` methods in Database class
- when constructing list of waited classes for email notifications, do not add class to list if its corresponding course is disabled
- add `is_course_disabled `flag to disable/enable course switches and show/hide disabled label on course page

**Disable Course Subscriptions**

https://user-images.githubusercontent.com/63625700/130334728-51104cb3-39e1-4070-bb33-33439f582a8a.mov

**Enable Course Subscriptions**

https://user-images.githubusercontent.com/63625700/130334791-5492a378-2aa7-459a-b67f-635abf98bdd5.mov

### CODE REVIEW

File: src/database.py [hunk 1]
```diff
@@ -181,6 +181,30 @@ def get_current_section(self, netid, courseid):
...
+        course_data = self.get_course(courseid)
+        if not course_data:
+            raise Exception(f"{courseid} is not a valid course ID.")
+            return False
+        try:
+            self._db.admin.update_one({}, { "$addToSet": { "disabled_courses": courseid }})
+            return True
+        except:
+            raise Exception(f"Could not add {courseid} to list of disabled courses.")
+            return False
```
Comment (nicholaspad) [inline src/database.py:194]: disabling a courseID multiple times won't cause issues right? since it's a set?

File: src/database.py [hunk 2]
```diff
@@ -181,6 +181,30 @@ def get_current_section(self, netid, courseid):
...
+        course_data = self.get_course(courseid)
+        if not course_data:
+            raise Exception(f"{courseid} is not a valid course ID.")
+            return False
+        try:
+            self._db.admin.update_one({}, { "$pull": { "disabled_courses": courseid }})
+            return True
+        except:
+            raise Exception(f"Could not remove {courseid} from list of disabled courses.")
+            return False
```
Comment (nicholaspad) [inline src/database.py:206]: same as above

File: src/monitor.py [hunk 3]
```diff
@@ -25,9 +25,13 @@ def _construct_waited_classes(self):
 
         for class_ in waited_classes:
             classid = class_["classid"]
-            deptnum = self._db.classid_to_course_deptnum(classid)
+            deptnum, courseid = self._db.classid_to_course_info(classid)
 
-            print("adding classid", classid, "from", deptnum)
+            # skip sections whose course is disabled
+            if self._db.is_course_disabled(courseid):
+                continue
```
Comment (nicholaspad) [inline src/monitor.py:32]: clean!

File: src/database.py (hunk 1 above)
Comment (nicholaspad) [inline src/database.py:194]: disabling a courseID multiple times won't cause issues right? since it's a set?

@shannon-heh

**Review by nicholaspad — COMMENTED** [review]
thanks for doing this, it looks great :)

**Comment by nicholaspad** [comment]
thanks for the video tests as well :)

**Review by nicholaspad — APPROVED** [review]

File: src/monitor.py (hunk 3 above)
Comment (shannon-heh) [reply src/monitor.py:32]: xie

File: src/database.py (hunk 1 above)
Comment (shannon-heh) [reply src/database.py:194]: yes. on frontend, it'll just say "disabled successfully".

**Pull request merged**